addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -126,10 +132,18 @@ options.registry.SnippetPopup = options.Class.extend({
     _computeWidgetState: function (methodName, params) {
         switch (methodName) {
             case 'moveBlock':
-                return this.$target.closest('footer').length ? 'moveToFooter' : 'moveToBody';
+                return this.$target[0].closest('.s_popup').dataset.popupParent === 'footer' ?
+                    'moveToFooter' : 'moveToBody';
         }
         return this._super(...arguments);
     },
+    /**
+     * Dirties the wrap and the footer to ensure that they will be saved.
+     */
+    _dirtyPage() {
+        document.querySelector('#wrap').classList.add('o_dirty');
+        document.querySelector('#footer').classList.add('o_dirty');
+    },
```

Comment:
Why would you need that (it supposed to be automatic) and why would you not comment why you need it?